content/configuration/executor.md

```diff
@@ -0,0 +1,147 @@
+      restrictAccess: false
+```
+
+The default configuration for Helm chart can be found in [values.yaml](https://github.com/kubeshop/botkube/blob/main/helm/botkube/values.yaml).
+
+## Merging strategy
+
+BotKube takes into account only bindings for a given execution Namespace. For example:
+
+1. `@BotKube get po/botkube -n botkube` - collect `kubectl` executor bindings that include `botkube` or `all` Namespaces.
+2. `@BotKube get po -A` - collect all `kubectl` executor bindings that include `all` Namespace.
+3. `@BotKube get po` - first, we resolve the execution Namespace. For that, we collect all enabled `kubectl` executor bindings and check the `defaultNamespace` property. If property is not define, we use the `default` Namespace. With resolved execution Namespace, we run the logic define in the first step.
+
+For all collected `kubectl` executors, we merge properties with the following strategy:
+- `commands.verbs` - append
+- `commands.resources` - append
+- `commands.defaultNamespace` - override
+- `commands.restrictAccess` - override
+
+
+The order of the binding list is important as it impacts properties that are overridden. The priority is given to the last binding specified on the list.
+
+### Example
+
+Let's consider such configuration:
+```yaml
+communications:
+  'default-group':
+    slack:
+      channels:
+        'default':
+          name: "random"
+          bindings:
+            executors:
+              - kubectl-pod
+              - kubectl-wait
+              - kubectl-all-ns
+              - kubectl-exec
+
+executors:
+  'kubectl-pod':
+    kubectl:
+      enabled: true
+      namespaces:
+        include: [ "botkube", "default" ]
+      commands:
+        verbs: [ "get" ]
+        resources: [ "pods" ]
+      restrictAccess: false
+  'kubectl-wait':
+    kubectl:
+      enabled: true
+      namespaces:
+        include: [ "botkube", "default" ]
+      commands:
+        verbs: [ "wait" ]
+      restrictAccess: true
+  'kubectl-all-ns':
+    kubectl:
+      enabled: true
+      namespaces:
+        include:
+          - all
+      commands:
+        verbs: [ "get" ]
+        resources: [ "deployments" ]
+  'kubectl-exec':
+    kubectl:
+      enabled: false
+      namespaces:
+        include: [ "botkube", "default" ]
+      commands:
+        verbs: [ "exec" ]
+      restrictAccess: false
+```
+
+We can see that:
+- for `botkube` and `default` Namespaces we can get and wait for Pods. This is the result of merging `kubectl-pod` and `kubectl-wait`.
+- for all Namespaces we can only get Deployments as specified by `kubectl-all-ns`.
+- the `exec` command is not allowed as the `kubectl-exec` binding is disabled (`kubectl.enabled=false`). As a result is not taken into account.
+- the `kubectl` works in a restricted access because the `kubectl-wait` binding is the last one which is both enabled and sets the `restrictAccess` property to a particular value.
```

Comment:
If there is a period at the end of sentence:

```suggestion
We can see that:
- For `botkube` and `default` Namespaces, we can execute `get` and `wait` commands for Pods. This is the result of merging `kubectl-pod` and `kubectl-wait`.
- For all Namespaces we can execute `get` for Deployments, as specified by `kubectl-all-ns`.
- The `exec` command is not allowed as the `kubectl-exec` binding is disabled (`kubectl.enabled` is set to `false`).
- The `kubectl` works in a restricted access because the `kubectl-wait` binding is enabled and sets the `restrictAccess` property to a `true`. The `kubectl-wait` was specified as the last item of executor bindings, so it overrides values `restrictAccess: false` from other bindings.
```

Reply:
I applied changes for the first 3 points but I don't agree with the 4th one:
```
- For `botkube` and `default` Namespaces, we can execute `get` and `wait` commands for Pods. This is the result of merging `kubectl-pod` and `kubectl-wait`.
- For all Namespaces we can execute `get` for Deployments, as specified by `kubectl-all-ns`.
- The `exec` command is not allowed as the `kubectl-exec` binding is disabled (`kubectl.enabled` is set to `false`).
- The `kubectl` works in a restricted access because the `kubectl-wait` binding is the **last one** which is both enabled and sets the `restrictAccess` property to `true`.
```